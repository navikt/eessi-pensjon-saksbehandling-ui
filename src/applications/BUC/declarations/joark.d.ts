
export interface JoarkFileVariant {
  variantformat:string;
  filnavn: string;
}

export interface JoarkDoc {
  dokumentInfoId: string;
  tittel: string;
  dokumentvarianter: Array<JoarkFileVariant>;
}
export interface JoarkPoster {
  journalpostId: string;
  tittel: string;
  tema: string;
  datoOpprettet: string;
  tilleggsopplysninger?: string;
  dokumenter: Array<JoarkDoc>
}

export interface JoarkFile {
  journalpostId: string;
  tittel: string;
  tema: string;
  datoOpprettet: Date;
  dokumentInfoId: string;
  tilleggsopplysninger?: string;
  variant: JoarkFileVariant
}

interface Content {
  base64: string;
}

export interface JoarkFileWithContent extends JoarkFile {
  content: Content;
  name: string;
  size: number;
  mimetype: string;
}
