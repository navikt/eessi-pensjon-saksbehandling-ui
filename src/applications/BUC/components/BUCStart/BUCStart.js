import React, { useState, useEffect } from 'react'
import PT from 'prop-types'
import _ from 'lodash'
import { connect, bindActionCreators } from 'store'
import PsychoPanel from 'components/ui/Psycho/PsychoPanel'
import { Flatknapp, HjelpetekstAuto, Hovedknapp, Input, NavFrontendSpinner, Normaltekst, Row, Select, Systemtittel, Undertittel } from 'components/ui/Nav'
import MultipleSelect from 'components/ui/MultipleSelect/MultipleSelect'
import * as bucActions from 'actions/buc'
import * as uiActions from 'actions/ui'
import * as appActions from 'actions/app'
import { getDisplayName } from 'utils/displayName'

export const mapStateToProps = (state) => {
  return {
    currentBUC: state.buc.currentBUC,
    bucsInfo: state.buc.bucsInfo,
    buc: state.buc.buc,
    subjectAreaList: state.buc.subjectAreaList,
    bucList: state.buc.bucList,
    tagList: state.buc.tagList,
    rinaId: state.buc.rinaId,

    loading: state.loading,
    sakId: state.app.params.sakId,
    aktoerId: state.app.params.aktoerId,
    bucParam: state.app.params.buc,
    locale: state.ui.locale
  }
}

const mapDispatchToProps = (dispatch) => {
  return { actions: bindActionCreators(Object.assign({}, bucActions, appActions, uiActions), dispatch) }
}

const placeholders = {
  subjectArea: 'buc:form-chooseSubjectArea',
  buc: 'buc:form-chooseBuc'
}

const BUCStart = (props) => {
  const { sakId, aktoerId, rinaId, buc, bucParam, bucsInfo } = props
  const { subjectAreaList, bucList, tagList } = props
  const { t, actions, currentBUC, locale, loading, mode } = props

  // these values may be collected through a form, but if they are in URL, then they will be set
  const [_sakId, setSakId] = useState(sakId)
  const [_aktoerId, setAktoerId] = useState(aktoerId)
  const [_buc, setBuc] = useState(bucParam)

  const [_subjectArea, setSubjectArea] = useState('Pensjon')
  const [_tags, setTags] = useState([])
  const [validation, setValidation] = useState({})

  const [isBucCreated, setIsBucCreated] = useState(false)
  const [hasBucInfoSaved, setHasBucInfoSaved] = useState(false)

  useEffect(() => {
    if (currentBUC && !sakId) {
      actions.setStatusParam('sakId', currentBUC.casenumber)
    }
  }, [currentBUC, sakId, actions])

  useEffect(() => {
    if (!loading.verifyingCaseNumber && _.isEmpty(currentBUC) && sakId && aktoerId) {
      actions.verifyCaseNumber({
        sakId: sakId,
        aktoerId: aktoerId
      })
    }
  }, [currentBUC, actions, loading, sakId, aktoerId])

  useEffect(() => {
    if (!loading.verifyingCaseNumber && !_.isEmpty(currentBUC)) {
      if (subjectAreaList === undefined && !loading.gettingSubjectAreaList) {
        actions.getSubjectAreaList()
      }
      if (bucList === undefined && !loading.gettingBucList) {
        actions.getBucList()
      }
      if (tagList === undefined && !loading.gettingTagList) {
        actions.getTagList()
      }
    }
  }, [currentBUC, actions, loading, bucList, subjectAreaList, tagList])

  useEffect(() => {
    if (!isBucCreated && buc) {
      actions.saveBucsInfo({
        bucsInfo: bucsInfo,
        aktoerId: aktoerId,
        tags: _tags,
        buc: buc
      })
      setIsBucCreated(true)
    }
  }, [actions, loading, bucsInfo, aktoerId, buc, _buc, _tags, rinaId, isBucCreated])

  useEffect(() => {
    if (!hasBucInfoSaved && loading.savingBucsInfo) {
      setHasBucInfoSaved(true)
    }
    if (hasBucInfoSaved && !loading.savingBucsInfo && buc) {
      actions.setMode('sednew')
      setHasBucInfoSaved(false)
    }
  }, [actions, loading, buc, hasBucInfoSaved])

  const onSakIdChange = (e) => {
    resetValidationState('sakId')
    setSakId(e.target.value.trim())
  }

  const onAktoerIdChange = (e) => {
    resetValidationState('aktoerId')
    setAktoerId(e.target.value.trim())
  }

  const onVerifyCaseButtonClick = () => {
    if (!_sakId) {
      setValidationState('sakId', t('buc:validation-noSakId'))
    }
    if (!_aktoerId) {
      setValidationState('aktoerId', t('buc:validation-noAktoerId'))
    }
    if (hasNoValidationErrors()) {
      actions.verifyCaseNumber({
        sakId: _sakId,
        aktoerId: _aktoerId
      })
    }
  }

  const onForwardButtonClick = () => {
    validateSubjectArea(_subjectArea)
    validateBuc(_buc)
    if (hasNoValidationErrors()) {
      actions.createBuc(_buc)
    }
  }

  const onCancelButtonClick = () => {
    actions.resetBuc()
    actions.setMode('buclist')
  }

  const validateSubjectArea = (subjectArea) => {
    if (!subjectArea || subjectArea === placeholders.subjectArea) {
      setValidationState('subjectAreaFail', t('buc:validation-chooseSubjectArea'))
    } else {
      resetValidationState('subjectAreaFail')
    }
  }

  const validateBuc = (buc) => {
    if (!buc || buc === placeholders.buc) {
      setValidationState('bucFail', t('buc:validation-chooseBuc'))
    } else {
      resetValidationState('bucFail')
    }
  }

  const resetValidationState = (_key) => {
    setValidation(_.omitBy(validation, (value, key) => {
      return key === _key
    }))
  }

  const hasNoValidationErrors = () => {
    return _.isEmpty(validation)
  }

  const setValidationState = (key, value) => {
    setValidation({
      ...validation,
      [key]: value
    })
  }

  const onSubjectAreaChange = (e) => {
    const thisSubjectArea = e.target.value
    setSubjectArea(thisSubjectArea)
    validateSubjectArea(thisSubjectArea)
  }

  const onBucChange = (e) => {
    const thisBuc = e.target.value
    setBuc(thisBuc)
    validateBuc(thisBuc)
  }

  const onTagsChange = (tagsList) => {
    setTags(tagsList)
  }

  const renderOptions = (options, type) => {
    if (typeof options === 'string') {
      options = [options]
    }
    if (!options || Object.keys(options).length === 0) {
      options = [{
        key: placeholders[type],
        value: t(placeholders[type])
      }]
    }

    if (!options[0].key || (options[0].key && options[0].key !== placeholders[type])) {
      options.unshift({
        key: placeholders[type],
        value: t(placeholders[type])
      })
    }
    return options.map(el => {
      let key, value
      if (typeof el === 'string') {
        key = el
        value = el
      } else {
        key = el.key || el.navn
        value = el.value || el.navn
      }
      return <option value={key} key={key}>{getOptionLabel(value)}</option>
    })
  }

  const getOptionLabel = (value) => {
    let label = value
    let description = t('buc:buc-' + value.replace(':', '.'))
    if (description !== 'buc-' + value) {
      label += ' - ' + description
    }
    return label
  }

  const getSpinner = (text) => {
    return <div className='a-buc-c-bucstart__spinner ml-2'>
      <NavFrontendSpinner type='S' />
      <div className='float-right ml-2'>{t(text)}</div>
    </div>
  }

  const tagObjectList = tagList ? tagList.map(tag => {
    return {
      value: tag,
      label: t('buc:tag-' + tag)
    }
  }) : []

  const allowedToForward = () => {
    return _buc && _subjectArea && hasNoValidationErrors() && !loading.creatingBUC && !loading.savingBucsInfo
  }

  if (!currentBUC) {
    return <div className='a-buc-c-bucstart__form-for-sakid-and-aktoerid'>
      {mode === 'page' ? <div className='mb-5'>
        <PsychoPanel closeButton>{t('buc:help-startCase')}</PsychoPanel>
      </div> : null}
      <Row>
        <div className='col-md-6'>
          <Input aria-describedby='help-sakId'
            className='-buc-c-bucstart__sakid'
            id='a-buc-c-bucstart__sakid-input-id'
            label={t('buc:form-sakId')}
            value={_sakId || ''}
            bredde='fullbredde'
            onChange={onSakIdChange}
            feil={validation.sakId ? { feilmelding: t(validation.sakId) } : null} />
        </div>
      </Row>
      <Row>
        <div className='col-md-6'>
          <Input
            className='-buc-c-bucstart__aktoerid'
            id='a-buc-c-bucstart__aktoerid-input-id'
            label={t('buc:form-aktoerId')}
            value={_aktoerId || ''}
            bredde='fullbredde'
            onChange={onAktoerIdChange}
            feil={validation.aktoerId ? { feilmelding: t(validation.aktoerId) } : null} />
        </div>
      </Row>
      <Row className='mt-6'>
        <div className='col-md-12'>
          <Hovedknapp
            id='a-buc-c-bucstart__forward-button-id'
            className='a-buc-c-bucstart__forward-button'
            disabled={loading && loading.verifyingCaseNumber}
            spinner={loading && loading.verifyingCaseNumber}
            onClick={onVerifyCaseButtonClick}>
            {loading && loading.verifyingCaseNumber ? t('buc:loading-verifyingCaseNumber') : t('ui:search')}
          </Hovedknapp>
        </div>
      </Row>
    </div>
  }

  return <React.Fragment>
    {mode === 'page' ? <React.Fragment>
      <Systemtittel className='mb-4'>{t('buc:app-startCaseDescription')}</Systemtittel>
      <div className='mb-5'>
        <PsychoPanel closeButton>{t('help-startCase2')}</PsychoPanel>
      </div>
    </React.Fragment> : null}
    <Row className='mb-3'>
      <div className='col-md-6 pr-3'>
        <Select
          className='a-buc-c-bucstart__subjectarea-list flex-fill'
          id='a-buc-c-bucstart__subjectarea-select-id'
          aria-describedby='help-subjectArea'
          bredde='fullbredde'
          feil={validation.subjectAreaFail ? { feilmelding: validation.subjectAreaFail } : null}
          label={<div className='label'>
            <span>{t('buc:form-subjectArea')}</span>
            <HjelpetekstAuto id='help-subjectArea'>
              {t('buc:help-subjectArea')}
            </HjelpetekstAuto>
          </div>}
          value={_subjectArea || []}
          onChange={onSubjectAreaChange}>
          {renderOptions(subjectAreaList, 'subjectArea')}
        </Select>
        <Select
          className='a-buc-c-bucstart__buc flex-fill'
          id='a-buc-c-bucstart__buc-select-id'
          aria-describedby='help-buc'
          bredde='fullbredde'
          feil={validation.bucFail ? { feilmelding: validation.bucFail } : null}
          label={<div className='label'>
            <span>{t('buc:form-buc')}</span>
            <HjelpetekstAuto id='help-buc'>
              {t('buc:help-buc')}
            </HjelpetekstAuto>
          </div>}
          value={_buc || placeholders.buc}
          onChange={onBucChange}>
          {renderOptions(bucList, 'buc')}
        </Select>
      </div>
      <div className='col-md-6 pl-3'>
        <div className='flex-fill'>
          <Undertittel className='mb-2'>{t('buc:form-tagsForBUC')}</Undertittel>
          <div className='mb-3'>
            <Normaltekst className='mb-2'>{t('buc:form-tagsForBUC-description')}</Normaltekst>
            <MultipleSelect
              className='a-buc-c-bucstart__tags flex-fill'
              id='a-buc-c-bucstart__tags-select-id'
              placeholder={t('buc:form-tagPlaceholder')}
              aria-describedby='help-tags'
              locale={locale}
              value={_tags}
              hideSelectedOptions={false}
              onChange={onTagsChange}
              optionList={tagObjectList} />
          </div>
        </div>

        <div className='selectBoxMessage mt-2 mb-2'>{!loading ? null
          : loading.gettingSubjectAreaList ? getSpinner('buc:loading-subjectArea')
            : loading.gettingBucList ? getSpinner('buc:loading-buc') : null}
        </div>
      </div>
    </Row>
    <Row className='mb-3'>
      <div className='col-md-12'>
        <Hovedknapp
          id='a-buc-BUCStart-forward-button'
          className='forwardButton'
          disabled={!allowedToForward()}
          spinner={loading.creatingBUC}
          onClick={onForwardButtonClick}>
          {loading.creatingBUC ? t('buc:loading-creatingCaseinRINA')
            : loading.savingBucsInfo ? t('buc:loading-savingBucInfo')
              : t('buc:form-createCaseinRINA')}
        </Hovedknapp>
        <Flatknapp
          id='a-buc-BUCStart-cancel'
          className='cancelButton ml-2'
          onClick={onCancelButtonClick}>{t('ui:cancel')}</Flatknapp>
      </div>
    </Row>
  </React.Fragment>
}

BUCStart.propTypes = {
  currentBUC: PT.object,
  mode: PT.string,
  actions: PT.object,
  loading: PT.object,
  t: PT.func,
  subjectAreaList: PT.array,
  bucList: PT.array,
  bucParam: PT.string,
  locale: PT.string.isRequired,
  sakId: PT.string,
  aktoerId: PT.string,
  rinaId: PT.string
}

const ConnectedBUCStart = connect(
  mapStateToProps,
  mapDispatchToProps
)(BUCStart)

ConnectedBUCStart.displayName = `Connect(${getDisplayName(BUCStart)})`

export default ConnectedBUCStart
