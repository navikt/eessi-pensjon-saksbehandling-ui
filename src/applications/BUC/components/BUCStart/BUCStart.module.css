.flexDiv {
  display: flex;
  align-items: flex-end;
  .flex-2 {
    flex: 2;
  }
}
