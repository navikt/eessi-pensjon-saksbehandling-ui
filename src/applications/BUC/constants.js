export const BUC_MODE = 'BUC/MODE'
