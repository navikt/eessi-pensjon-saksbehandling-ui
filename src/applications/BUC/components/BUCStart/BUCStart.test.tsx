import { createBuc, getBucOptions, getTagList, saveBucsInfo } from 'src/actions/buc'
import * as constants from 'src/constants/constants'
import { AllowedLocaleString } from 'src/declarations/app.d'
import { BucsInfo } from 'src/declarations/buc'
import {render, screen, fireEvent} from '@testing-library/react'
import { within } from '@testing-library/dom'
import _ from 'lodash'
import mockFeatureToggles from 'src/mocks/app/featureToggles'
import mockPerson from 'src/mocks/person/personPdl'
import mockPersonAvdods from 'src/mocks/person/personAvdod'
import mockBucOptions from 'src/mocks/buc/bucOptions'
import mockBucs from 'src/mocks/buc/bucs'
import mockBucsInfo from 'src/mocks/buc/bucsInfo'
import mockTagsList from 'src/mocks/buc/tagsList'
import { stageSelector } from 'src/setupTests'
import BUCStart, { BUCStartProps } from './BUCStart'
import React from 'react'

jest.mock('src/actions/buc', () => ({
  cleanNewlyCreatedBuc: jest.fn(),
  createBuc: jest.fn(),
  getBucOptions: jest.fn(),
  getTagList: jest.fn(),
  resetBuc: jest.fn(),
  saveBucsInfo: jest.fn()
}))

const bucs = _.keyBy(mockBucs(), 'caseId')
const currentBuc = _.values(bucs)[0].caseId
const mockPersonAvdod = mockPersonAvdods(1)

const defaultSelector = {
  bucOptions: mockBucOptions,
  bucParam: undefined,
  bucs,
  bucsInfo: {},
  currentBuc: undefined,
  featureToggles: mockFeatureToggles,
  loading: {},
  locale: 'nb' as AllowedLocaleString,
  newlyCreatedBuc: undefined,
  person: mockPerson,
  personAvdods: mockPersonAvdod,
  pesysContext: constants.VEDTAKSKONTEKST,
  sakId: '123',
  subjectAreaList: ['Pensjon'],
  tagList: mockTagsList
}

describe('applications/BUC/components/BUCStart/BUCStart', () => {
  /*let wrapper: any*/
  const initialMockProps: BUCStartProps = {
    aktoerId: '456',
    onBucCreated: jest.fn(),
    onBucCancelled: jest.fn()
  } as BUCStartProps

  beforeEach(() => {
    stageSelector(defaultSelector, {})
  })

  it('Render: Has proper HTML structure', () => {
    render(<BUCStart {...initialMockProps} />)
    expect(screen.getByTestId('a_buc_c_BUCStart')).toBeInTheDocument()
    expect(screen.getByTestId('a_buc_c_BUCStart--subjectarea-select-id')).toBeInTheDocument()
    expect(screen.getByTestId('a_buc_c_BUCStart--buc-select-id')).toBeInTheDocument()
    expect(screen.queryByTestId('a_buc_c_BUCStart--avdod-select-id')).not.toBeInTheDocument()
    expect(screen.getByText('buc:form-tagsForBUC-description'))
    expect(screen.getByTestId('a_buc_c_BUCStart--buttons-id')).toBeInTheDocument()
    expect(screen.getByTestId('a_buc_c_BUCStart--forward-button-id')).toBeInTheDocument()
    expect(screen.getByTestId('a_buc_c_BUCStart--cancel-button-id')).toBeInTheDocument()
  })

  it('Render: render Avdod when BUC is P_BUC_02', () => {
    render(<BUCStart {...initialMockProps} />)
    expect(screen.queryByTestId('a_buc_c_BUCStart--avdod-select-id')).not.toBeInTheDocument()
    const select = within(screen.getByTestId('a_buc_c_BUCStart--buc-select-id')).getByRole('combobox')
    fireEvent.keyDown(select, { key: 'ArrowDown' })
    fireEvent.keyDown(select, { key: 'ArrowDown' })
    fireEvent.keyDown(select, { key: 'Enter' })
    expect(screen.getByTestId('a_buc_c_BUCStart--avdod-select-id')).toBeInTheDocument()
  })

  it('UseEffect: Fetch buc list', () => {
    (getBucOptions as jest.Mock).mockReset()
    stageSelector(defaultSelector, {
      bucOptions: undefined
    })
    render(<BUCStart {...initialMockProps} />)
    expect(getBucOptions).toHaveBeenCalled()
  })

  it('UseEffect: Fetch tag list', () => {
    (getTagList as jest.Mock).mockReset()
    stageSelector(defaultSelector, {
      tagList: undefined
    })
    render(<BUCStart {...initialMockProps} />)
    expect(getTagList).toHaveBeenCalled()
  })

  it('UseEffect: set avdod if we have one, and _buc selected on P_BUC_02', () => {
    jest.useFakeTimers({ legacyFakeTimers: true })
    const mockUseState = jest.fn()
    const useStateSpy = jest.spyOn(React, 'useState').mockImplementation(() => {
      return [undefined, mockUseState]
    })
    stageSelector(defaultSelector, {
      bucParam: 'P_BUC_02',
      personAvdods: mockPersonAvdods(1)
    })
    render(<BUCStart {...initialMockProps} />)
    jest.runAllImmediates()

    // TODO
    // expect(mockUseState).toHaveBeenCalled()
    useStateSpy.mockRestore()
  })

  it('UseEffect: saves bucsInfo after when buc was saved', () => {
    (saveBucsInfo as jest.Mock).mockReset()
    stageSelector(defaultSelector, {
      bucsInfo: mockBucsInfo as BucsInfo,
      bucs,
      currentBuc,
      newlyCreatedBuc: bucs[currentBuc]
    })
    render(<BUCStart {...initialMockProps} initialIsCreatingBuc />)
    expect(saveBucsInfo).toHaveBeenCalledWith({
      bucsInfo: mockBucsInfo,
      aktoerId: '456',
      tags: [],
      buc: _.values(bucs)[0]
    })
  })

  it('UseEffect: call bucCreated after bucInfo is saved', () => {
    (initialMockProps.onBucCreated as jest.Mock).mockReset()
    stageSelector(defaultSelector, {
      newlyCreatedBuc: bucs[currentBuc]
    })
    render(<BUCStart {...initialMockProps} initialCreatingBucInfo />)
    expect(initialMockProps.onBucCreated).toHaveBeenCalled()
  })

  it('Handling: invalid onForwardButtonClick(): nothing selected', () => {
    render(<BUCStart {...initialMockProps} />)
    expect(screen.queryByText('message:error-validationbox-bucstart')).not.toBeInTheDocument()
    fireEvent.click(screen.getByTestId('a_buc_c_BUCStart--forward-button-id'))
    expect(screen.getByText('message:error-validationbox-bucstart')).toBeInTheDocument()
    screen.debug()
  })

  it('Handling: invalid onForwardButtonClick(): set warning for P_BUC_02 and no avdod', () => {
    stageSelector(defaultSelector, {
      personAvdods: []
    })
    render(<BUCStart {...initialMockProps} />)
    expect(screen.queryByTestId('a_buc_c_BUCStart--warning-id')).not.toBeInTheDocument()
    const select = within(screen.getByTestId('a_buc_c_BUCStart--buc-select-id')).getByRole('combobox')
    fireEvent.keyDown(select, { key: 'ArrowDown' })
    fireEvent.keyDown(select, { key: 'ArrowDown' })
    fireEvent.keyDown(select, { key: 'Enter' })
    fireEvent.click(screen.getByTestId('a_buc_c_BUCStart--forward-button-id'))
    expect(screen.getByTestId('a_buc_c_BUCStart--warning-id')).toBeInTheDocument()
    expect(screen.getByTestId('a_buc_c_BUCStart--warning-id')).toHaveTextContent(
      'Advarsel' + 'message:alert-noDeceased'
    )
  })

  it('Handling: valid onForwardButtonClick()', () => {
    // select P_BUC_02
    stageSelector(defaultSelector, {
    })
    render(<BUCStart {...initialMockProps} />)
    const select = within(screen.getByTestId('a_buc_c_BUCStart--buc-select-id')).getByRole('combobox')
    fireEvent.keyDown(select, { key: 'ArrowDown' })
    fireEvent.keyDown(select, { key: 'ArrowDown' })
    fireEvent.keyDown(select, { key: 'Enter' })
    fireEvent.click(screen.getByTestId('a_buc_c_BUCStart--forward-button-id'))
    expect(createBuc).toHaveBeenCalledWith({
      buc: 'P_BUC_02',
      avdod: mockPersonAvdod![0]
    })
  })

    it('Handling: onCancelButtonClick()', () => {
      (initialMockProps.onBucCancelled as jest.Mock).mockReset()
      render(<BUCStart {...initialMockProps} />)
      fireEvent.click(screen.getByTestId('a_buc_c_BUCStart--cancel-button-id'))
      expect(initialMockProps.onBucCancelled).toHaveBeenCalled()
    })
})
