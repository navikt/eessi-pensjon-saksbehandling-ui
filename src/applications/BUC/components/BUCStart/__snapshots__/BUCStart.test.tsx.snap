// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`applications/BUC/components/BUCStart/BUCStart Render: match snapshot 1`] = `
.c10 .skjemaelement--feilmelding .multipleSelect--control {
  border: 1px solid;
}

ui-fss.g4[id="c9"] {
  content: "c10,";
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-left: -0.5rem;
  margin-right: -0.5rem;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

ui-fss.g12[id="c0"] {
  content: "c1,";
}

.c3 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  margin-left: 0.5rem;
  margin-right: 0.5rem;
}

ui-fss.g16[id="c2"] {
  content: "c3,";
}

.c8 {
  display: inline-block;
  margin-left: 2rem;
}

.c11 {
  display: inline-block;
  margin-left: 1rem;
}

ui-fss.g22[id="c7"] {
  content: "c8,c11,";
}

.c5 {
  margin-bottom: 2rem;
}

.c6 {
  margin-bottom: 1rem;
}

ui-fss.g23[id="c4"] {
  content: "c5,c6,";
}

@media (min-width:768px) {
  .c1 {
    -webkit-flex-direction: row;
    -ms-flex-direction: row;
    flex-direction: row;
  }
}

<div
  data-testid="a_buc_c_BUCStart"
>
  <div
    class="c0 c1"
  >
    <div
      class="c2 c3"
    >
      <div
        class="c4 c5"
        size="2"
      />
      <label
        class="navds-text-field--label navds-label"
      >
        buc:form-chooseSubjectArea
      </label>
      <div
        class=""
        data-testid="a_buc_c_BUCStart--subjectarea-select-id"
      >
        <div
          class=" css-91gbow-container"
          id="a_buc_c_BUCStart--subjectarea-select-id"
        >
          <span
            class="css-1f43avz-a11yText-A11yText"
            id="react-select-2-live-region"
          />
          <span
            aria-atomic="false"
            aria-live="polite"
            aria-relevant="additions text"
            class="css-1f43avz-a11yText-A11yText"
          />
          <div
            class=" css-1q6azai-control"
          >
            <div
              class=" css-319lph-ValueContainer"
            >
              <div
                class=" css-173nacq-singleValue"
              >
                Pensjon
              </div>
              <div
                class=" css-6j8wv5-Input"
                data-value=""
              >
                <input
                  aria-autocomplete="list"
                  aria-expanded="false"
                  aria-haspopup="true"
                  autocapitalize="none"
                  autocomplete="off"
                  autocorrect="off"
                  class=""
                  id="a_buc_c_BUCStart--subjectarea-select-id"
                  role="combobox"
                  spellcheck="false"
                  style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                  tabindex="0"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class=" css-1hb7zxy-IndicatorsContainer"
            >
              <span
                class=" css-1pm5dtp-indicatorSeparator"
              />
              <div
                aria-hidden="true"
                class=" css-tlfecz-indicatorContainer"
              >
                <svg
                  aria-hidden="true"
                  class="css-tj5bde-Svg"
                  focusable="false"
                  height="20"
                  viewBox="0 0 20 20"
                  width="20"
                >
                  <path
                    d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
      <div
        class="c4 c6"
      />
      <label
        class="navds-text-field--label navds-label"
      >
        buc:form-chooseBuc
      </label>
      <div
        class=""
        data-testid="a_buc_c_BUCStart--buc-select-id"
      >
        <div
          class=" css-91gbow-container"
          id="a_buc_c_BUCStart--buc-select-id"
        >
          <span
            class="css-1f43avz-a11yText-A11yText"
            id="react-select-3-live-region"
          />
          <span
            aria-atomic="false"
            aria-live="polite"
            aria-relevant="additions text"
            class="css-1f43avz-a11yText-A11yText"
          />
          <div
            class=" css-1q6azai-control"
          >
            <div
              class=" css-319lph-ValueContainer"
            >
              <div
                class=" css-14el2xx-placeholder"
                id="react-select-3-placeholder"
              />
              <div
                class=" css-6j8wv5-Input"
                data-value=""
              >
                <input
                  aria-autocomplete="list"
                  aria-describedby="react-select-3-placeholder"
                  aria-expanded="false"
                  aria-haspopup="true"
                  autocapitalize="none"
                  autocomplete="off"
                  autocorrect="off"
                  class=""
                  id="a_buc_c_BUCStart--buc-select-id"
                  role="combobox"
                  spellcheck="false"
                  style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                  tabindex="0"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class=" css-1hb7zxy-IndicatorsContainer"
            >
              <span
                class=" css-1pm5dtp-indicatorSeparator"
              />
              <div
                aria-hidden="true"
                class=" css-tlfecz-indicatorContainer"
              >
                <svg
                  aria-hidden="true"
                  class="css-tj5bde-Svg"
                  focusable="false"
                  height="20"
                  viewBox="0 0 20 20"
                  width="20"
                >
                  <path
                    d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="c7 c8"
      size="2"
    />
    <div
      class="c2 c3"
    >
      <div
        class="c4 c5"
        size="2"
      />
      <div
        class="c9 c10"
      >
        <label
          class="navds-text-field--label navds-label"
          for="a_buc_c_BUCStart--tags-select-id"
        >
          <label
            class="navds-text-field--label navds-label"
          >
            buc:form-tagsForBUC
          </label>
          <div
            class="c4 c6"
          />
          <p
            class="navds-body-long"
          >
            buc:form-tagsForBUC-description
          </p>
        </label>
        <div
          class="multipleSelect css-91gbow-container"
        >
          <span
            class="css-1f43avz-a11yText-A11yText"
            id="react-select-4-live-region"
          />
          <span
            aria-atomic="false"
            aria-live="polite"
            aria-relevant="additions text"
            class="css-1f43avz-a11yText-A11yText"
          />
          <div
            class="multipleSelect__control css-1q6azai-control"
          >
            <div
              class="multipleSelect__value-container multipleSelect__value-container--is-multi css-319lph-ValueContainer"
            >
              <div
                class="multipleSelect__placeholder css-14el2xx-placeholder"
                id="react-select-4-placeholder"
              />
              <div
                class="multipleSelect__input-container css-6j8wv5-Input"
                data-value=""
              >
                <input
                  aria-autocomplete="list"
                  aria-describedby="react-select-4-placeholder"
                  aria-expanded="false"
                  aria-haspopup="true"
                  aria-label="buc:form-tagsForBUC"
                  autocapitalize="none"
                  autocomplete="off"
                  autocorrect="off"
                  class="multipleSelect__input"
                  id="a_buc_c_BUCStart--tags-select-id"
                  role="combobox"
                  spellcheck="false"
                  style="opacity: 1; width: 100%; grid-area: 1 / 2; min-width: 2px; border: 0px; margin: 0px; outline: 0; padding: 0px;"
                  tabindex="0"
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class="multipleSelect__indicators css-1hb7zxy-IndicatorsContainer"
            >
              <span
                class="multipleSelect__indicator-separator css-141yg44-indicatorSeparator"
              />
              <div
                aria-hidden="true"
                class="multipleSelect__indicator multipleSelect__dropdown-indicator css-tlfecz-indicatorContainer"
              >
                <svg
                  aria-hidden="true"
                  class="css-tj5bde-Svg"
                  focusable="false"
                  height="20"
                  viewBox="0 0 20 20"
                  width="20"
                >
                  <path
                    d="M4.516 7.548c0.436-0.446 1.043-0.481 1.576 0l3.908 3.747 3.908-3.747c0.533-0.481 1.141-0.446 1.574 0 0.436 0.445 0.408 1.197 0 1.615-0.406 0.418-4.695 4.502-4.695 4.502-0.217 0.223-0.502 0.335-0.787 0.335s-0.57-0.112-0.789-0.335c0 0-4.287-4.084-4.695-4.502s-0.436-1.17 0-1.615z"
                  />
                </svg>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="c4 c5"
    size="2"
  />
  <div
    data-testid="a_buc_c_BUCStart--buttons-id"
  >
    <button
      class="navds-button navds-button--primary navds-button--medium"
      data-amplitude="buc.new.create"
      data-testid="a_buc_c_BUCStart--forward-button-id"
    >
      <span
        aria-live="polite"
        class="navds-label"
      >
        buc:form-createCaseinRINA
      </span>
    </button>
    <div
      class="c7 c11"
    />
    <button
      class="navds-button navds-button--tertiary navds-button--medium"
      data-amplitude="buc.new.cancel"
      data-testid="a_buc_c_BUCStart--cancel-button-id"
    >
      <span
        aria-live="polite"
        class="navds-label"
      >
        ui:cancel
      </span>
    </button>
  </div>
  <div
    class="c4 c6"
  />
</div>
`;
